internal/controller/deployment/controller_watch.go

```diff
@@ -58,6 +58,22 @@ func (r *Reconciler) setupDeploymentArtifactRefIndex(ctx context.Context, mgr ct
 	)
 }
 
+func (r *Reconciler) setupEndpointsOwnerRefIndex(ctx context.Context, mgr ctrl.Manager) error {
+	return mgr.GetFieldIndexer().IndexField(
+		ctx,
+		&choreov1.Endpoint{},
+		"metadata.ownerReferences",
+		func(rawObj client.Object) []string {
+			endpoint := rawObj.(*choreov1.Endpoint)
```

Comment:
Shall we use the cast check similar to https://github.com/openchoreo/openchoreo/blob/main/internal/controller/deployment/controller_watch.go#L51?

Ideally, this should not happen. But I think it is safe to avoid unnecessary panic.